laplace/utils/matrix.py

```diff
@@ -145,14 +148,16 @@ def _bmm(self, W: torch.Tensor) -> torch.Tensor:
                 Q = Fs[0]
                 p = len(Q)
                 W_p = W[:, cur_p:cur_p+p].T
-                SW.append((Q @ W_p).T)
+                SW.append((Q @ W_p).T if Q.ndim > 1 else (Q.view(-1, 1) * W_p).T)
```

Comment:
Why the else statement although `Q = torch.eye()` for `ndim == 1`? I think in this case you could make Q above a vector of ones and save the memory?

Reply:
Again, it's @runame change.

Reply:
`Q.ndim` \neq `Hi.ndim`, but I don't see any case where `Q.ndim == 1` anymore. `Q` is not a vector for diagonal Kronecker factors, because it is not necessarily the identity but some permutation matrix. The code should probably be rewritten to avoid explicitly building the permutation matrix in memory, but this requires duplicating this code in the if-else statement to deal with the diagonal Kronecker factor case separately. Btw, the PR with the changes is [here](https://github.com/aleximmer/Laplace/pull/136).

Reply:
Let's leave this be for now then. We can revisit this when we add SINGD baseline. Might be useful then.